### ISSUE TITLE
Question: Is there support for Sauce Labs?

### ISSUE THREAD

**Issue by frbk** [BorisOsipov/wdio-reportportal-reporter#108]
I noticed that version 5 of reportportal added Sauce Labs support by adding attributes to tests and was wondering if reporter supports it? I cant find a way to update attributes per test. Am I missing something?

**Comment by BorisOsipov** [comment]
>if reporter supports it?

Now not.

>I cant find a way to update attributes per test

I didn't add such api because didn't recieved requests :)

To move forward could you clarify some questions:

1)
In [docs](https://reportportal.io/docs/Sauce-Labs-integration)
>XXXXXXXX - # of job in Sauce Labs

Is jod id equal `browser.sessionId`. See  wdio [docs](https://webdriver.io/docs/browserobject.html#get-desired-capabilities)?

2)

Can I run tests on sauselab and local grid\browsers simultaneously?
I think if all tests run on sauselab I can automatically assign jobid to all of them...

**Comment by frbk** [comment]
Hey @BorisOsipov,
Thanks for getting back to me. Yes, job id is `browser.sessionId`. You cant run test on local grid\browsers when using Sauce Labs integration.

**Comment by frbk** [comment]
Do you need anything else?

**Comment by BorisOsipov** [comment]
Seems implementation in report portal part is still not completed(See conversation in report portal slack below)

So I am not sure it is good to add it now... What do you think?

> 
> Boris Osipov Feb 28th at 9:06 AM
> Hi folks! In v5 docs described sauselab support. are there any client lib which such integration support?
> Boris Osipov  23 hours ago
> I am interesting in some reference implementation in any language. Which java agent can I take? junit 4/5 or testng?
> 
> ghdmitry  22 hours ago
> @Boris_Osipov we decided not to share this info for now.
> we still plan some updates for UI part, in order to make it more smooth.
> so will make it public once ready from UI part as well.
> 
> Boris Osipov  22 hours ago
> but you already added info in docs) (edited) 
> 
> Boris Osipov  22 hours ago
> and guys ask about how can they use it
> 
> ghdmitry  22 hours ago
> we did. but it doesnt mean they are ready for contribution )

**Comment by frbk** [comment]
Interesting. Did they mention what needs to be done? I though that their current implementation only required you add `key:value` to a test. (I tried this and it worked https://github.com/reportportal/reportportal/issues/348). Is it possible to pass attributes to each test using the reporter right now? I noticed you are parsing test tags but I can't find anything else.

**Comment by BorisOsipov** [comment]
>Is it possible to pass attributes to each test using the reporter right now?

Nope. And I still don't know how to make it not so ugly as it is in allure reporter.See example:
```
describe('Suite', () => {
    it('test', () => {
        allureReporter.addFeature('Feature')
    })
})
```

So you will have to add smth like that in each(!!!) test.
```
    it('test', () => {
        reportPortal.addAttribute('SLID', 'FOOBAR')
		//actual test content
    })
```
It is ugly and ridiculous from my point of view.

What do you think if we move forward with next approach:

1. I'll add new config flag e.g. `sauseLabTest`
2. If flag is `true` I will automatically assign `SLID` by current `browser.sessionId` for each test item in report portal. 
Btw it may not work for `multiremote` wdio feature, but I think is it good enough for start.

what do you think?

**Comment by frbk** [comment]
I like your flag idea. We can start off with that and worry about `multiremote` later. Thanks for getting back to me.

**Comment by BorisOsipov** [comment]
@frbk can you try with wdio-reportportal-reporter@6.0.1-alpha?

Need setup in config:
```
const conf = {
    reportPortalClientConfig: {
	// ...
    },
	// ...
    isSauseLabRun: true,
	// ...
};
```
and requires wdio-reportportal-service@6.0.0.

**Comment by frbk** [comment]
Hey @BorisOsipov,
Had a chance to test it out and it works great.
Thank you.

**Comment by BorisOsipov** [comment]
@frbk ok. will merge to master and release soon.

**Comment by honza164** [comment]
HI @BorisOsipov,
is this feature in master already pls? 
doesnt work for me with :\
```
"wdio-reportportal-reporter": "^6.0.0",
"wdio-reportportal-service": "^6.0.0"
```
thx!

**Comment by BorisOsipov** [comment]
@honza164 nope.

Please, use `wdio-reportportal-reporter@6.0.1-alpha`.

Before merging it must be fixed for case when `browser.reloadSession()` used - in such a case seems `SLID` will be reported incorrectly.

**Comment by BorisOsipov** [comment]
Released in 6.1.0

**Comment by Shivin89** [comment]
> Released in 6.1.0



> @honza164 nope.
> 
> Please, use `wdio-reportportal-reporter@6.0.1-alpha`.
> 
> Before merging it must be fixed for case when `browser.reloadSession()` used - in such a case seems `SLID` will be reported incorrectly.

-------------------------------------

Hi Boris ,

I am using the Cucumber6 plugin with Report Portal, and they have added the feature of passing attribute Key & value pair in Step Definition (Test Steps), But my question here is How I will pass my job id automatically in test steps?
Session id ? How will I pass that ?

**Comment by BorisOsipov** [comment]
@Shivin89 fixed in #140
